tests/nvidiagpu/deploy-gpu-test.go

```diff
@@ -398,21 +501,83 @@ var _ = Describe("GPU", Ordered, Label(tsparams.LabelSuite), func() {
+					glog.V(gpuparams.GpuLogLevel).Infof("The GPU packagemanifest '%s' was not "+
+						"found in the default '%s' catalog", gpuPackage, gpuCatalogSourceDefault)
+
+					if createGPUCustomCatalogsource {
+						glog.V(gpuparams.GpuLogLevel).Infof("Creating custom catalogsource '%s' for GPU Operator, "+
+							"with index image '%s'", gpuCustomCatalogSource, gpuCustomCatalogsourceIndexImage)
+
+						glog.V(gpuparams.GpuLogLevel).Infof("Deploying a custom GPU catalogsource '%s' with '%s' "+
+							"index image", gpuCustomCatalogSource, gpuCustomCatalogsourceIndexImage)
+
+						gpuCustomCatalogSourceBuilder := olm.NewCatalogSourceBuilderWithIndexImage(inittools.APIClient,
+							gpuCustomCatalogSource, gpuCatalogSourceNamespace, gpuCustomCatalogsourceIndexImage,
+							gpuCustomCatalogSourceDisplayName, gpuCustomCatalogSourcePublisherName)
+
+						Expect(gpuCustomCatalogSourceBuilder).NotTo(BeNil(), "Failed to Initialize "+
+							"CatalogSourceBuilder for custom GPU catalogsource '%s'", gpuCustomCatalogSource)
+
+						createdGPUCustomCatalogSourceBuilder, err := gpuCustomCatalogSourceBuilder.Create()
+						glog.V(gpuparams.GpuLogLevel).Infof("Creating custom GPU Catalogsource builder object "+
+							"'%s'", createdGPUCustomCatalogSourceBuilder.Definition.Name)
+						Expect(err).ToNot(HaveOccurred(), "error creating custom GPU catalogsource "+
+							"builder Object name %s:  %v", gpuCustomCatalogSource, err)
+
+						By("Sleep for 30 seconds to allow the GPU custom catalogsource to be created")
+						time.Sleep(30 * time.Second)
 
-				glog.V(gpuparams.GpuLogLevel).Infof("The gpu packagemanifest name returned: %s",
-					gpuPkgManifestBuilderByCatalog.Object.Name)
+						glog.V(gpuparams.GpuLogLevel).Infof("Wait up to 4 mins for custom GPU catalogsource " +
+							"to be ready")
+
+						Expect(createdGPUCustomCatalogSourceBuilder.IsReady(4 * time.Minute)).NotTo(BeFalse())
+
+						gpuCatalogSource = createdGPUCustomCatalogSourceBuilder.Definition.Name
+
+						glog.V(gpuparams.GpuLogLevel).Infof("Custom GPU catalogsource '%s' is now ready",
+							createdGPUCustomCatalogSourceBuilder.Definition.Name)
+
+						gpuPkgManifestBuilderByCustomCatalog, err := olm.PullPackageManifestByCatalog(inittools.APIClient,
+							gpuPackage, gpuCatalogSourceNamespace, gpuCustomCatalogSource)
+
+						Expect(err).ToNot(HaveOccurred(), "error getting GPU packagemanifest '%s' "+
+							"from custom catalog '%s':  %v", gpuPackage, gpuCustomCatalogSource, err)
+
+						By("Get the GPU Default Channel from Packagemanifest")
+						gpuDefaultSubscriptionChannel = gpuPkgManifestBuilderByCustomCatalog.Object.Status.DefaultChannel
+						glog.V(gpuparams.GpuLogLevel).Infof("GPU channel '%s' retrieved from packagemanifest "+
+							"of custom catalogsource '%s'", gpuDefaultSubscriptionChannel, gpuCustomCatalogSource)
+
+					} else {
+						Skip("gpu-operator-certified packagemanifest not found in default 'certified-operators'" +
```

Comment:
Same question here. Shouldn't we fail if the GPU operator can't be deployed? No bundle, not in the default catalog, no alternative catalog.